modules/virt-configuring-vm-disk-sharing.adoc

```diff
@@ -0,0 +1,58 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virtual_disks/virt-configuring-shared-volumes-for-vms.adoc
+
+:_content-type: PROCEDURE
+[id="virt-configuring-vm-disk-sharing{context}"]
+= Configuring disk sharing by using virtual machine disks
+
+You can configure block volumes so that multiple virtual machines (VMs) can share storage.
+
+The application running in the guest determines the storage option you must configure for the VM. A disk of type `disk` exposes the volume as an ordinary disk to the VM.
+
+.Prerequisites
+
+* The volume access mode must be `ReadWriteMany` (RWX) if the VMs that are sharing disks are running on different nodes.
++
+If the VMs that are sharing disks are running on the same node, `ReadWriteOnce` (RWO) volume access mode is sufficient.
+
+* The storage provider must support the required Container Storage Interface (CSI) driver.
+
+.Procedure
+
+. Create the `VirtualMachine` manifest for your VM to set the required values, as shown in the following example:
+
++
+[source,yaml]
+====
+----
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: <vm_name>
+spec:
+ volumemode: Block
```

Comment:
Hi @alicefr, could you please look, is it a doc bug, or we need to add this field to `oc explain`?

```
oc explain VirtualMachine.spec.volumemode
...
error: field "volumemode" does not exist
```

Reply:
Not sure it's the issue, there's no volumeMode in VirtualMachine.spec
volumeMode can be in **PersistentVolumeClaim** spec

```
$ oc explain VirtualMachine.spec
GROUP:      kubevirt.io
KIND:       VirtualMachine
VERSION:    v1

FIELD: spec <Object>

DESCRIPTION:
    Spec contains the specification of VirtualMachineInstance created
    
FIELDS:
  dataVolumeTemplates	<[]Object>
    dataVolumeTemplates is a list of dataVolumes that the VirtualMachineInstance
    template can reference. DataVolumes in this list are dynamically created for
    the VirtualMachine and are tied to the VirtualMachine's life-cycle.

  instancetype	<Object>
    InstancetypeMatcher references a instancetype that is used to fill fields in
    Template

  liveUpdateFeatures	<Object>
    LiveUpdateFeatures references a configuration of hotpluggable resources

  preference	<Object>
    PreferenceMatcher references a set of preference that is used to fill fields
    in Template

  runStrategy	<string>
    Running state indicates the requested running state of the
    VirtualMachineInstance mutually exclusive with Running

  running	<boolean>
    Running controls whether the associatied VirtualMachineInstance is created
    or not Mutually exclusive with RunStrategy

  template	<Object> -required-
    Template is the direct specification of VirtualMachineInstance

```

Reply:
This is a [typo](https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#disk-sharing). Should be `volumeMode`. Also, the indent needs to be fixed.

@aspauldi Please check your indents very carefully. A bad indent breaks the YAML.

Reply:
@jpeimer you are right, `volumeMode` is for PVCs. I might have missed this during the review. Thanks!
@aspauldi the correct example is in the [kubevirt guide](https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#disk-sharing)

Reply:
@jpeimer @alicefr, Should I add something in place of volumeMode?

Reply:
The `volumeMode` is for PVC but you don't have it in your example. So, I think you can simply remove it. On the VM specification there isn't necessary